src/wallet/wallet2.cpp

```diff
@@ -5035,27 +5035,27 @@ std::string wallet2::make_uri(const std::string &address, const std::string &pay
   }
 
   std::string uri = "monero:" + address;
-  bool n_fields = 0;
+  int n_fields = 0;
 
   if (!payment_id.empty())
   {
-    uri += (n_fields++ ? "&" : "?") + std::string("tx_payment_id=") + payment_id;
+    uri += (n_fields++ > 0 ? "&" : "?") + std::string("tx_payment_id=") + payment_id;
```

Comment:
Those changes are a bit pointless, no ? They just halve the range in which the code will work :) Not that the range change matters in practice, though. Might as well remove them, and maybe make n_fields unsigned while at it.